plugin/sycl/objective/regression_obj.cc

```diff
@@ -0,0 +1,192 @@
+  float scale_pos_weight;
+  // declare parameters
+  DMLC_DECLARE_PARAMETER(RegLossParam) {
+    DMLC_DECLARE_FIELD(scale_pos_weight).set_default(1.0f).set_lower_bound(0.0f)
+      .describe("Scale the weight of positive examples by this factor");
+  }
+};
+
+template<typename Loss>
+class RegLossObj : public ObjFunction {
+ protected:
+  HostDeviceVector<int> label_correct_;
+
+ public:
+  RegLossObj() = default;
+
+  void Configure(const std::vector<std::pair<std::string, std::string> >& args) override {
+    param_.UpdateAllowUnknown(args);
+    qu_ = device_manager.GetQueue(ctx_->Device());
+  }
+
+  void GetGradient(const HostDeviceVector<bst_float>& preds,
+                   const MetaInfo &info,
+                   int iter,
+                   HostDeviceVector<GradientPair>* out_gpair) override {
+  if (info.labels.Size() == 0U) {
+    LOG(WARNING) << "Label set is empty.";
+  }
+  CHECK_EQ(preds.Size(), info.labels.Size())
+      << " " << "labels are not correctly provided"
+      << "preds.size=" << preds.Size() << ", label.size=" << info.labels.Size() << ", "
+      << "Loss: " << Loss::Name();
+
+  size_t const ndata = preds.Size();
+  out_gpair->Resize(ndata);
+
+  // TODO: add label_correct check
+  label_correct_.Resize(1);
+  label_correct_.Fill(1);
+
+  bool is_null_weight = info.weights_.Size() == 0;
+
+  ::sycl::buffer<bst_float, 1> preds_buf(preds.HostPointer(), preds.Size());
+  ::sycl::buffer<bst_float, 1> labels_buf(info.labels.Data()->HostPointer(), info.labels.Size());
+  ::sycl::buffer<GradientPair, 1> out_gpair_buf(out_gpair->HostPointer(), out_gpair->Size());
+  ::sycl::buffer<bst_float, 1> weights_buf(is_null_weight ? NULL : info.weights_.HostPointer(),
+                                         is_null_weight ? 1    : info.weights_.Size());
+
+  const size_t n_targets = std::max(info.labels.Shape(1), static_cast<size_t>(1));
+
+  auto scale_pos_weight = param_.scale_pos_weight;
+  if (!is_null_weight) {
+    CHECK_EQ(info.weights_.Size(), info.labels.Shape(0))
+      << "Number of weights should be equal to number of data points.";
+  }
+
+  int flag = 1;
+  {
+    ::sycl::buffer<int, 1> additional_input_buf(&flag, 1);
+    qu_.submit([&](::sycl::handler& cgh) {
+        auto preds_acc            = preds_buf.get_access<::sycl::access::mode::read>(cgh);
+        auto labels_acc           = labels_buf.get_access<::sycl::access::mode::read>(cgh);
+        auto weights_acc          = weights_buf.get_access<::sycl::access::mode::read>(cgh);
+        auto out_gpair_acc        = out_gpair_buf.get_access<::sycl::access::mode::write>(cgh);
+        auto additional_input_acc = additional_input_buf.get_access<::sycl::access::mode::write>(cgh);
+        cgh.parallel_for<>(::sycl::range<1>(ndata), [=](::sycl::id<1> pid) {
+          int idx = pid[0];
+          bst_float p = Loss::PredTransform(preds_acc[idx]);
+          bst_float w = is_null_weight ? 1.0f : weights_acc[idx/n_targets];
```

Comment:
integer division here in the indexing. Usually that would want me to make sure I'm doing the right conversion. Worth a double check. Is ndata's idx per the 1st dimension. is idx supposed to be valued info.labels.Shape(1) spacing between value or is it like [1,2,3,4] etc.